src/Files.App.Storage/Storables/HomeFolder/HomeFolder.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Files Community
+// Licensed under the MIT License.
+
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using Windows.Win32;
+
+namespace Files.App.Storage.Storables
+{
+	public partial class HomeFolder : IHomeFolder
+	{
+		public string Id => "Home"; // Will be "files://Home" in the future.
+
+		public string Name => "Home";
+
+		/// <inheritdoc/>
+		public async IAsyncEnumerable<IStorableChild> GetItemsAsync(StorableType type = StorableType.Folder, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+		{
+			await foreach (var folder in GetQuickAccessFolderAsync(cancellationToken))
+			{
+				cancellationToken.ThrowIfCancellationRequested();
+
+				yield return folder;
+			}
+
+			await foreach (var drive in GetLogicalDrivesAsync(cancellationToken))
+			{
+				cancellationToken.ThrowIfCancellationRequested();
+
+				yield return drive;
+			}
+
+			await foreach (var location in GetNetworkLocationsAsync(cancellationToken))
+			{
+				cancellationToken.ThrowIfCancellationRequested();
+
+				yield return location;
+			}
+		}
+
+		/// <inheritdoc/>
+		public IAsyncEnumerable<IStorableChild> GetQuickAccessFolderAsync(CancellationToken cancellationToken = default)
+		{
+			IFolder folder = new WindowsFolder(new Guid("3936e9e4-d92c-4eee-a85a-bc16d5ea0819"));
+			return folder.GetItemsAsync(StorableType.Folder, cancellationToken);
+		}
+
+		/// <inheritdoc/>
+		public async IAsyncEnumerable<IStorableChild> GetLogicalDrivesAsync([EnumeratorCancellation] CancellationToken cancellationToken = default)
+		{
+			var availableDrives = PInvoke.GetLogicalDrives();
+			if (availableDrives is 0)
+				yield break;
+
+			int count = BitOperations.PopCount(availableDrives);
+			var driveLetters = new char[count];
+
+			count = 0;
+			char driveLetter = 'A';
+			while (availableDrives is not 0)
+			{
+				if ((availableDrives & 1) is not 0)
+					driveLetters[count++] = driveLetter;
+
+				availableDrives >>= 1;
+				driveLetter++;
+			}
+
+			foreach (int letter in driveLetters)
+			{
+				cancellationToken.ThrowIfCancellationRequested();
+
+				if (WindowsStorable.TryParse($"{letter}:\\") is not IWindowsStorable driveRoot)
+					throw new InvalidOperationException();
+
+				yield return new WindowsFolder(driveRoot.ThisPtr);
+				await Task.Yield();
+			}
+
+		}
+
+		/// <inheritdoc/>
+		public IAsyncEnumerable<IStorableChild> GetNetworkLocationsAsync(CancellationToken cancellationToken = default)
+		{
+			Guid FOLDERID_NetHood = new("{C5ABBF53-E17F-4121-8900-86626FC2C973}");
```

Comment:
What are these ids?

Reply:
Its a shell folder GUID called `FOLDERID_NetHood` for network locations. We've used this here:

https://github.com/files-community/Files/blob/7728bda269e3a251451993aa6b4e9ffc2001662d/src/Files.App/Services/Storage/StorageNetworkService.cs#L97